ddev/ddev
# Feature showing messages to the users, fixes #3799

## The Issue

* #3799 

We like to show messages to the user e.g. a hint to support DDEV with sponsoring the project.

## How This PR Solves The Issue

This patch introduces an easy way to provide infos, warnings and rotating ticker messages in the remote-config repository which will be regulary downloaded and shown to the user.

Open tasks:
* [x] define a good first message about sponsoring DDEV
* [x] define some first tips

## Manual Testing Instructions

To test this feature run a ddev command e.g. `ddev describe`, messages should be shown. Manipulate or delete the state file `~/.ddev/.state.yaml` to get faster results. It's also possible to change the interval of the ticker in the `~/.ddev/global_config.yaml` or switch to the testing branch of the remote-config, description to do so can be found in the config file.

## Automated Testing Overview



## Related Issue Link(s)

## Release/Deployment Notes





<a href="https://gitpod.io/#https://github.com/ddev/ddev/pull/4885"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
